libavcodec/qsvdec.c

```diff
@@ -336,6 +337,10 @@ static int qsv_decode_header(AVCodecContext *avctx, QSVContext *q,
         return ret;
 
     param->mfx.CodecId = ret;
+    video_signal_info.Header.BufferId = MFX_EXTBUFF_VIDEO_SIGNAL_INFO;
+    video_signal_info.Header.BufferSz = sizeof(video_signal_info);
+    param->ExtParam    = header_ext_params;
```

Comment:
Can `param->ExtParam` already contain something here (maybe from `qsv_decode_preinit`)?

Reply:
Yes, it can contain something here, hence it will be reset to q->ext_buffers after MFXVideoDECODE_DecodeHeader. 
Note MFX_EXTBUFF_VIDEO_SIGNAL_INFO is only used in MFX_EXTBUFF_VIDEO_SIGNAL_INFO() and other ext buffer might result in error in MFXVideoDECODE_DecodeHeader().  So we can't append MFX_EXTBUFF_VIDEO_SIGNAL_INFO to param->ExtParam.